package.yaml

```diff
@@ -18,69 +14,90 @@ extra-doc-files:
 extra-source-files:
   - package.yaml
 
+language: GHC2021
+
 ghc-options:
   - -Weverything
   - -Wno-all-missed-specialisations
   - -Wno-missed-specialisations
   - -Wno-missing-exported-signatures # re-enables missing-signatures
   - -Wno-missing-import-lists
+  - -Wno-missing-kind-signatures
   - -Wno-missing-local-signatures
+  - -Wno-missing-safe-haskell-mode
   - -Wno-monomorphism-restriction
+  - -Wno-partial-fields
+  - -Wno-prepositive-qualified-module
   - -Wno-safe
   - -Wno-unsafe
-
-when:
-  - condition: "impl(ghc >= 9.2)"
-    ghc-options:
-      - -Wno-missing-kind-signatures
-  - condition: "impl(ghc >= 8.10)"
-    ghc-options:
-      - -Wno-missing-safe-haskell-mode
-      - -Wno-prepositive-qualified-module
+  - -fwrite-ide-info
 
 dependencies:
   - base < 5
 
 default-extensions:
-  - BangPatterns
-  - DataKinds
+  - AllowAmbiguousTypes
   - DeriveAnyClass
-  - DeriveFoldable
-  - DeriveFunctor
-  - DeriveGeneric
-  - DeriveLift
-  - DeriveTraversable
   - DerivingStrategies
-  - FlexibleContexts
-  - FlexibleInstances
+  - DuplicateRecordFields
+  - FunctionalDependencies
   - GADTs
-  - GeneralizedNewtypeDeriving
+  - ImpredicativeTypes
   - LambdaCase
-  - MultiParamTypeClasses
+  - NoFieldSelectors
   - NoImplicitPrelude
-  - NoMonomorphismRestriction
+  - OverloadedRecordDot
   - OverloadedStrings
-  - RankNTypes
+  - QuantifiedConstraints
   - RecordWildCards
-  - ScopedTypeVariables
-  - StandaloneDeriving
-  - TypeApplications
   - TypeFamilies
+  - UndecidableInstances
 
 library:
   source-dirs: library
+  dependencies:
+    - internal
+
+internal-libraries:
```

Comment:
I to am excited for this feature, but I've heard it can break some tooling. You've found it works OK?

Reply:
The only issue I've had is that it seems to break the Stack+HLS combination in particular, so I ended up configuring HLS to load via Cabal.